### PR TITLE
cli speed optimizations - imports2

### PR DESCRIPTION
**Pull request by Equanox**
This PR adds a import statement to get rid of find(). It is a breaking change as child Bobfiles must now be declared instead of automatically resolved.

### CODE REVIEW

File: bob/bobfile/bobfile.go [hunk 1]
```diff
@@ -105,7 +107,7 @@ func bobfileRead(dir string) (_ *Bobfile, err error) {
 	bobfilePath := filepath.Join(dir, global.BobFileName)
 
 	if !file.Exists(bobfilePath) {
-		return nil, ErrBobfileNotFound
+		return nil, ErrBobfileNotFound //fmt.Errorf("%s contains no Bobfile, %w", dir, ErrBobfileNotFound)
```
Comment (rdnt) [inline bob/bobfile/bobfile.go:110]: Cleanup

File: bobtask/input.go [hunk 2]
```diff
@@ -51,10 +53,16 @@ func (t *Task) filteredInputs() ([]string, error) {
 			ignores = append(ignores, list...)
 			continue
 		}
+
+		// TODO: when "*" is passed as input it's likely to hit the cache
+		// ass there is no further information. Think how to handle the cache correctly
```
Comment (rdnt) [inline bobtask/input.go:58]: Typo..

File: bobtask/sanitize.go [hunk 3]
```diff
@@ -77,13 +79,25 @@ var absPathMap = make(map[string]absolutePathOrError, 10000)
...
+			abs = filepath.Join(opts.wd, path)
+		} else {
+			abs, err = filepath.Abs(path)
+			if err != nil {
+				return "", err
+			}
+		}
+
 	}
+	// fmt.Println(" to " + abs)
```
Comment (rdnt) [inline bobtask/sanitize.go:100]: Cleanup

File: bob/playground.go [hunk 4]
```diff
@@ -417,6 +422,8 @@ func createPlaygroundBobfileSecondLevel(dir string, overwrite bool, projectName
 	bobfile.Version = "1.2.3"
 	bobfile.Project = projectName
 
+	bobfile.Imports = []string{"third-level"}
```
Comment (rdnt) [inline bob/playground.go:425]: Neat. 👌🏻

**Review by rdnt — APPROVED** [review]
Some cleanup is required in general, but works nicely. Speed hasn't changed much but will hopefully be much better for larger projects.

**Comment by Equanox** [comment]
@rdnt The speed improvement is mostly relevant for auto completion (`time bob gettask`). 

Medium size project:
Before: 0m0,195s
After:    0m0,033s

**Pull request merged**